storage/src/main/java/io/delta/storage/HDFSLogStore.java

```diff
@@ -0,0 +1,174 @@
+ * </ol>
+ */
+public class HDFSLogStore extends HadoopFileSystemLogStore {
+
+    public static final String NO_ABSTRACT_FILE_SYSTEM_EXCEPTION_MESSAGE = "No AbstractFileSystem";
+
+    public HDFSLogStore(Configuration hadoopConf) {
+        super(hadoopConf);
+    }
+
+    @Override
+    public void write(
+            Path path,
+            Iterator<String> actions,
+            Boolean overwrite,
+            Configuration hadoopConf) throws IOException {
+        final boolean isLocalFs = path.getFileSystem(hadoopConf) instanceof RawLocalFileSystem;
+        if (isLocalFs) {
+            // We need to add `synchronized` for RawLocalFileSystem as its rename will not throw an
+            // exception when the target file exists. Hence we must make sure `exists + rename` in
+            // `writeInternal` for RawLocalFileSystem is atomic in our tests.
+            synchronized(this) {
+                writeInternal(path, actions, overwrite, hadoopConf);
+            }
+        } else {
+            // rename is atomic and also will fail when the target file exists. Not need to add the extra
+            // `synchronized`.
+            writeInternal(path, actions, overwrite, hadoopConf);
+        }
+    }
+
+    @Override
+    public Boolean isPartialWriteVisible(Path path, Configuration hadoopConf) {
+        return true;
+    }
+
+    /**
+     * @throws UnsupportedFileSystemException if the file system from the default configuration is
+     *                                        not supported
+     */
+    protected FileContext getFileContext(
+            Path path,
+            Configuration hadoopConf) throws UnsupportedFileSystemException {
+        return FileContext.getFileContext(path.toUri(), hadoopConf);
+    }
+
+    /**
+     * Create a temporary path (to be used as a copy) for the input {@code path}
+     */
+    protected Path createTempPath(Path path) {
+        return new Path(
+            path.getParent(),
+            String.format(".%s.%s.tmp", path.getName(), UUID.randomUUID())
+        );
+    }
+
+    /**
+     * @throws IOException if this HDFSLogStore is used to write into a Delta table on a non-HDFS
+     *                     storage system.
+     * @throws FileAlreadyExistsException if {@code overwrite} is false and the file at {@code path}
+     *                                    already exists.
+     */
+    private void writeInternal(
+            Path path,
+            Iterator<String> actions,
+            Boolean overwrite,
+            Configuration hadoopConf) throws IOException {
+        final FileContext fc;
+        try {
+            fc = getFileContext(path, hadoopConf);
+        } catch (IOException e) {
+            if (e.getMessage().contains(NO_ABSTRACT_FILE_SYSTEM_EXCEPTION_MESSAGE)) {
+                final IOException newException = DeltaErrors.incorrectLogStoreImplementationException(e);
+                // TODO: logError(newException.getMessage(), newException.getCause())
```

Comment:
We can define a log object such as 
```
import org.slf4j.Logger
import org.slf4j.logger.LoggerFactory

  private static final Logger LOG = LoggerFactory.getLogger(HDFSLogStore.class);

```
and then use it to log.

Reply:
sg